### ISSUE TITLE
Add trigger information to criteria

### ISSUE THREAD

**Issue by danieldg** [i3/i3#4168]
## I'm submitting a…
<pre>
[ ] Bug
[x] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior
for_window expressions are matched against windows whenever any property changes

## Desired Behavior
allow wiriting a for_window expression that only matches when a window is initially mapped (or similar)

syntax: <pre>for_window [trigger="map" workspace="1"] (command)</pre>

## Example implementation in sway
https://github.com/swaywm/sway/pull/5609

**Comment by Airblader** [comment]
I definitely understand the rationale for this and would welcome such a feature. However, I'd like to start simple rather than lots of values for different properties having changed.

I'd also like to not call it "map" because windows being mapped is very technical speak most users are unfamiliar with.

**Comment by danieldg** [comment]
I agree renaming "map" to something more user-friendly like "create" would be sensible. The various values for trigger are intended to match the keys of criteria that are being changed, so there shouldn't be much confusion from having multiple names. However, I personally haven't found a good use for anything other than trigger="map"/"create" so changing this feature to just be a boolean [on_create] would also work.

**Comment by Airblader** [comment]
API-wise I'd stick with the string value as it can be extended, I just wouldn't want to implement too much right away and start with that single value. If a need arises for others it'd be easy enough to support.

**Comment by danieldg** [comment]
Ah. At least for the implementation in sway, it was just as easy to add all the various values as to only add map, but whatever is easier to do in i3 sounds good to me.

**Comment by budRich** [comment]
I propose using: `change="new"` instead of `trigger="map"` , to make it consistent with f.i. the output from: `i3-msg -mt subscribe '["window"]'` . I guess all events that change can be, could be used. This feature being accepted is great news!

**Comment by danieldg** [comment]
> I propose using: `change="new"` instead of `trigger="map"` , to make it consistent with f.i. the output from: `i3-msg -mt subscribe '["window"]'` . I guess all events that change can be, could be used. This feature being accepted is great news!

If `new` is already used in subscribe, that seems like a reasonable name for the value.  `change` seems like it might be a bit generic; maybe `onchange`? This is really just bike-shedding, the point is getting the ability.

**Comment by budRich** [comment]
Yeah. Other change values that could later be used are: "focus", "title" and "close". Sure "onchange" is more clear, but i guess it would then be named: "on_change", to be consistent with other criteria ("con_mark" e.t.c). Worth to consider is that this change/onchange thing will only be used with for_window commands. for_window kind of implies "on" imho.  

Maybe it is even better to have something like this:  
```  
for_window [on_new class=Firefox] ...
for_window [class=Firefox on_close] ...
```   

I think that i3ipc-python uses similar lingo.
